tests/components/fan/test_dyson.py

```diff
@@ -91,6 +93,29 @@ def _add_device(devices):
         self.hass.data[dyson.DYSON_DEVICES] = [device_fan, device_non_fan]
         dyson.setup_platform(self.hass, None, _add_device)
 
+    def test_get_state_attributes(self):
+        """Test async added to hass."""
+        device = _get_device_on()
+        component = dyson.DysonPureCoolLinkDevice(self.hass, device)
+        attributes = component.state_attributes
+        assert dyson.ATTR_IS_NIGHT_MODE in attributes
+        assert dyson.ATTR_IS_AUTO_MODE in attributes
+        assert ATTR_SPEED in attributes
+        assert ATTR_SPEED_LIST in attributes
+        assert ATTR_OSCILLATING in attributes
+
+    def test_async_added_to_hass(self):
+        """Test async added to hass."""
+        loop = asyncio.new_event_loop()
+
+        async def run():
+            """Test async function."""
+            device = _get_device_on()
+            component = dyson.DysonPureCoolLinkDevice(self.hass, device)
+            await component.async_added_to_hass()
+            assert device.add_message_listener.called
+        loop.run_until_complete(run())
```

Comment:
It's better to let the home assistant core set up the platform and add an entity. The core will then call `async_added_to_hass` on the entity, and we can assert after that.

Use `setup_component` to set up the component/platform.